handlers/dimension/listHandler.go

```diff
@@ -0,0 +1,26 @@
+package dimension
+
+import (
+	"encoding/json"
+	"net/http"
+	"github.com/ONSdigital/dp-dd-api-stub/models"
+	"fmt"
+)
+
+func listHandler(w http.ResponseWriter, req *http.Request) {
+	datasetID := req.URL.Query().Get(":datasetId")
+	dimensionID := req.URL.Query().Get(":dimensionId")
+
+	dimension, err := findDimension(datasetID, dimensionID)
+	if err != nil {
+		fmt.Println(err)
+		response, _ := json.Marshal(models.Dimension{})
+		w.WriteHeader(400)
+		w.Write(response)
+		return
+	}
+
+	jsonEncoder := json.NewEncoder(w)
+	jsonEncoder.Encode(dimension)
+	w.WriteHeader(200)
```

Comment:
WriteHeader needs to be before writing any data, i.e. before `jsonEncoder.Encode`

this is where `json.NewEncoder(w)` falls apart slightly - since we have to write the status code first, if there's an error encoding the value to JSON, we can't change the status code to a `500`

worth considering `json.Marshal()` instead, even though it's less efficient, it'll give more consistent API output